lib/poller.js

```diff
@@ -134,9 +135,18 @@ exports.do = async function (
   for (let i = 1; i <= maxPolls; i++) {
     const pollStep = (pollC * i) ** 2
     console.log(pollStep)
-    await timer(Math.min(pollStep, start + timeout - Date.now()))
-    if (Date.now() - start >= timeout) failOnTimeout(timeout, uuid)
+    const t = initialTimeout || timeout
+
+    if (!initialTimeout) {
+      await util.timer(Math.min(pollStep, start + t - Date.now()))
+    }
+    if (Date.now() - start >= timeout) failOnTimeout(t, uuid)
+
     const res = await ping(uuid, accessToken, apiUrl)
     if (res) return res
+    initialTimeout = null
   }
+  /* eslint-disable no-throw-literal */
+  throw `Polling failed after ${maxPolls} retries.`
```

Comment:
I think we want the time in there not the number of polls. The time is what a user cares about. Recall that we are encouraging the user (or tool like truffle-security) to make a reasonable guess and reward in response the closer they get this guess right.

Reply:
Great! Let's do it!